packages/wonder-blocks-badge/src/components/badge.tsx

```diff
@@ -1,5 +1,151 @@
+import {addStyle, AriaProps, StyleType} from "@khanacademy/wonder-blocks-core";
+import {styles as typographyStyles} from "@khanacademy/wonder-blocks-typography";
+import {border, semanticColor, sizing} from "@khanacademy/wonder-blocks-tokens";
+import {StyleSheet} from "aphrodite";
 import * as React from "react";
 
-export const Badge = () => {
-    return <div>Badge component</div>;
+export type BadgeProps = AriaProps & {
+    /**
+     * The label to display in the badge.
+     */
+    label?: string;
+    /**
+     * The icon to display in the badge. It can be a PhosphorIcon, a custom svg,
+     * or `img` element.
+     * Considerations:
+     * - If the icon conveys meaning, set the alt text on the icon being used
+     * - If the icon is an `img` element, it may need width: 100% and height: 100%
+     *   to render properly in the badge.
+     */
+    icon?: React.ReactElement;
+    /**
+     * The id for the badge.
+     */
+    id?: string;
+    /**
+     * The test id for the badge.
+     */
+    testId?: string;
+    /**
+     * Custom styles for the elements in the Badge component.
+     * - `root`: Styles the root element
+     * - `icon`: Styles the icon element
+     */
+    styles?: {
+        root?: StyleType;
+        icon?: StyleType;
+    };
 };
+
+const StyledDiv = addStyle("div");
+const StyledSpan = addStyle("span");
+
+/**
+ * Badges are visual indicators used to display concise information, such as
+ * a status, label, or count.
+ */
+const Badge = React.forwardRef<HTMLDivElement, BadgeProps>(function Badge(
+    props: BadgeProps,
+    ref: React.ForwardedRef<HTMLDivElement>,
+) {
+    const {icon, label, id, testId, styles: stylesProp, ...otherProps} = props;
+    if (!label && !icon) {
+        return <React.Fragment />;
+    }
+    return (
+        <StyledDiv
+            id={id}
+            data-testid={testId}
+            ref={ref}
+            style={[
+                typographyStyles.LabelXSmall,
+                styles.badge,
+                styles.defaultBadgeStyling,
+                icon && !label ? styles.iconOnly : {},
+                stylesProp?.root,
+            ]}
+            {...otherProps}
+        >
+            {icon && (
+                <StyledSpan
+                    style={[
+                        styles.icon,
+                        styles.defaultIconStyling,
+                        stylesProp?.icon,
+                    ]}
+                >
+                    {icon}
+                </StyledSpan>
+            )}
+            {label}
+        </StyledDiv>
+    );
+});
+
+export {Badge};
+
+const badgeTokens = {
```

Comment:
Setting up a structure for tokens related to badges. Let me know if anyone has any feedback on this - I was trying to follow the [naming guide](https://khanacademy.atlassian.net/wiki/x/FwB47)!

This doesn't reflect the TB styling yet. We can make updates to the tokens used (or if we need to have component themes) once the TB theme is in place! We may also make more updates to use `semantic core` tokens

Reply:
Thanks for the examples! I will set up the theming structure and token usage in a follow up PR!

Reply:
Sounds great!